File: app/src/main/java/ims/chat/application/ImsApplication.java
package ims.chat.application;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;
import cn.jpush.im.android.api.JMessageClient;

/**
 * 自定义Application
 */
public class ImsApplication extends Application {
    public static ImsApplication imsApplication;
    private static   Context context;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();

        JMessageClient.setDebugMode(true);
        JMessageClient.init(this);
        context = getApplicationContext();

    }

    public static Context getContext(){
        return context;
    }
}

File: app/src/main/java/ims/yang/com/ims/ImsApplication.kt
package ims.yang.com.ims

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context

class ImsApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        mApplication = this
        context = applicationContext
    }

    companion object {
        lateinit var mApplication: Application
        @SuppressLint("StaticFieldLeak")
        var context: Context? = null
            private set
    }

}
